File: WikiCodeParser/ParseResult.cs
using System.Collections.Generic;
using WikiCodeParser.Nodes;

namespace WikiCodeParser
{
    public class ParseResult
    {
        public INode Content { get; set; }

        public ParseResult()
        {
            Content = new NodeCollection();
        }

        public List<KeyValuePair<string, object>> GetMetadata()
        {
            var list = new List<KeyValuePair<string, object>>();
            Content.Walk(n =>
            {
                if (n is MetadataNode md) list.Add(new KeyValuePair<string, object>(md.Key, md.Value));
            });
            return list;
        }

        public string ToHtml() => Content.ToHtml();
        public string ToPlainText() => Content.ToPlainText();
    }
}

File: php/src/ParseResult.php
<?php

namespace LogicAndTrick\WikiCodeParser;

use LogicAndTrick\WikiCodeParser\Nodes\INode;
use LogicAndTrick\WikiCodeParser\Nodes\MetadataNode;
use LogicAndTrick\WikiCodeParser\Nodes\NodeCollection;
use LogicAndTrick\WikiCodeParser\Nodes\NodeExtensions;

class ParseResult
{
    public INode $content;

    public function __construct()
    {
        $this->content = new NodeCollection();
    }

    /** @noinspection PhpArrayUsedOnlyForWriteInspection */
    public function GetMetadata(): array
    {
        $list = [];
        NodeExtensions::Walk($this->content, function (INode $n) use ($list) {
            if ($n instanceof MetadataNode) $list[] = ['key' => $n->key, 'value' => $n->value];
            return true;
        });
        return $list;
    }

    public function ToHtml(): string
    {
        return $this->content->ToHtml();
    }

    public function ToPlainText(): string
    {
        return $this->content->ToPlainText();
    }
}
